app/js/actions/artifacts.js

```diff
@@ -7,11 +14,78 @@ export const expectingArtifact = () => ({
     type: 'EXPECTING_ARTIFACT'
 });
 
-export const deleteArtifact = (uuid) => ({
+export const hiddenDeleteArtifact = (uuid) => ({
     type: 'DELETE_ARTIFACT',
     uuid
 });
 
-export const refreshArtifacts = () => ({
-    type: 'REFRESH_ARTIFACTS'
-});
+export const deleteArtifact = (uuid) => {
+    return (dispatch, getState) => {
+        const { artifacts: { artifacts },
+                connection: { uri, availableApis, secretKey }
+        } = getState();
+        const httpVerb = 'DELETE';
+        const timestamp = Date.now();
+        const artifact = artifacts.filter(a => a.uuid === uuid)[0];
+        const body = JSON.stringify({
+            artifact
+        });
+        const digest = actions.makeB64Digest(secretKey, httpVerb, timestamp, body);
+        fetch(`http://${uri.split('/')[0]}${availableApis[0]}${artifact.uri}`, {
+            method: httpVerb,
+            headers: new Headers({
+                Authorization: `HMAC-SHA256 ${digest}`,
+                'Content-Type': 'application/json',
+                'X-QIIME-Timestamp': timestamp
+            }),
+            body
+        })
+        .then((response) => {
+            if (!response.ok) {
+                throw Error(response.statusText);
+            }
+            return response.json();
+        })
+        .then((json) => {
+            if (json.success) {
+                dispatch(hiddenDeleteArtifact(uuid));
+            }
+        });
+    };
+};
+
+export const loadArtifacts = () => {
+    return (dispatch, getState) => {
+        const { connection: { uri, availableApis } } = getState();
+        fetch(`http://${uri.split('/')[0]}${availableApis[0]}artifacts`, {
+            method: 'GET'
+        })
+        .then((response) => (response.json()))
+        .then((json) => {
+            json.artifacts.map(artifact => (
+                dispatch(newArtifact(artifact))
+            ));
+        });
+    };
+};
+
+export const refreshArtifacts = () => {
+    return (dispatch, getState) => {
+        const { artifacts, connection: { uri, availableApis } } = getState();
+        fetch(`http://${uri.split('/')[0]}${availableApis[0]}artifacts`, {
```

Comment:
``http://${uri.split('/')[0]}${availableApis[0]}SOMEPATH`` is starting to get tedious, we should factor that out at some point, we can probably also drop `availableApis` as we don't need to worry about backwards compatibility in a single distribution (Electron)